File: src/chapter_1_1_2.java
public class chapter_1_1_2 {

    public static void main(String args[]) {


        System.out.println("fib(1) = " + getFibNumber1(1));
        System.out.println("fib(2) = " + getFibNumber1(2));
        System.out.println("fib(3) = " + getFibNumber1(3));
        System.out.println("fib(10) = " + getFibNumber1(10));

        System.out.println("--- ---");

        System.out.println("fib(1) = " + getFibNumber2(1));
        System.out.println("fib(2) = " + getFibNumber2(2));
        System.out.println("fib(3) = " + getFibNumber2(3));
        System.out.println("fib(10) = " + getFibNumber2(10));
    }

    private static int getFibNumber1(int n) {
        if (n <= 0) return -1;
        if (n == 1) return 1;
        if (n == 2) return 2;
        int[] dp = new int[n + 1];
        dp[0] = 0;
        dp[1] = 1;
        dp[2] = 2;
        for (int i = 3; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n];
    }

    private static int getFibNumber2(int n) {
        if (n <= 0) return -1;
        if (n == 1) return 1;
        if (n == 2) return 2;

        int pre = 1, cur = 2;
        for (int i = 3; i <= n; i++) {
            int sum = pre + cur;
            pre = cur;
            cur = sum;
        }
        return cur;
    }

}

File: src/chapter_1_1_2.kt
fun main(args: Array<String>) {

    println("getFib(1) = " + getFib(1))
    println("getFib(2) = " + getFib(2))
    println("getFib(3) = " + getFib(3))
    println("getFib(10) = " + getFib(10))
}

fun getFib(n: Int): Int = when (n) {
    1 -> 1
    2 -> 2
    else -> {
        var pre = 1
        var cur = 2
        for (i in 3..n) {
            val sum = pre + cur
            pre = cur
            cur = sum
        }
        cur
    }
}